.github/workflows/deploy_and_test.yml

```diff
@@ -0,0 +1,61 @@
+name: CI
+
+on:
+  workflow_dispatch:
+  push:
+    branches: [ main ]
+    paths-ignore:
+      - "*.md"
+      - "*.png"
+      - "*.gif"
+      - "*.svg"
+      - "docs/**"
+      - ".vscode/**"
+  pull_request:
+    branches: [ main ]
+    paths-ignore:
+      - "*.md"
+      - "*.png"
+      - "*.gif"
+      - "*.svg"
+      - "docs/**"
+      - ".vscode/**"
+
+jobs:
+  build-and-test:
+      name: python ${{ matrix.python-version }}
+      runs-on: windows-latest
+      strategy:
+        fail-fast: false
+        max-parallel: 4
+        matrix:
+          python-version: ['3.6']
+      steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          path: scripts_src
+      - name: Setup python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Pip cache
+        uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-{{ matrix.python-version }}
+          restore-keys: |
+            ${{ runner.os }}-pip
+      - name: Pip install
+        working-directory: scripts_src
+        run: pip install .
+      - name: Test
+        working-directory: scripts_src
+        run: |
+          pip install pytest
```

Comment:
what about having a tracked `requirements-test.txt` which is specific about the version to use (perhaps as defined via `pip-compile`)

Reply:
good suggestion, thanks